plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/TransactionLogAccess.java

```diff
@@ -85,18 +90,21 @@
 
 public class TransactionLogAccess
 {
-    private static final Logger log = Logger.get(TransactionLogAccess.class);
-
     private final TypeManager typeManager;
     private final CheckpointSchemaManager checkpointSchemaManager;
     private final FileFormatDataSourceStats fileFormatDataSourceStats;
     private final TrinoFileSystemFactory fileSystemFactory;
     private final ParquetReaderOptions parquetReaderOptions;
-    private final Cache<CacheKey, TableSnapshot> tableSnapshots;
-    private final Cache<CacheKey, DeltaLakeDataFileCacheEntry> activeDataFileCache;
     private final boolean checkpointRowStatisticsWritingEnabled;
     private final int domainCompactionThreshold;
 
+    private final Cache<TableLocation, TableSnapshot> tableSnapshots;
+    private final Cache<TableVersion, DeltaLakeDataFileCacheEntry> activeDataFileCache;
+
+    // TODO move to query-level state
+    private final Map<QueriedLocation, Long> queriedLocations = new ConcurrentHashMap<>();
```

Comment:
Does `DeltaLakeMetadata#cleanupQuery` execute also when the query is aborted?
I'm concerned that, in unexpected situations, these maps (`queriedLocations` , `queriedTables`) will contain zombie objects putting pressure on the memory.

Reply:
yes, it should be

from the method docs

> This is the very last notification after the query finishes, whether it succeeds or fails.